### PR TITLE
[IMP] hr_attendance : fixing and improving of attendance onboarding

### PR DESCRIPTION
**Pull request by mepe-odoo**
Some texts are changed to be more comprehensible.
The mobile view was reworked to avoid broken onboarding view.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.js [hunk 1]
```diff
@@ -1,59 +1,131 @@
...
+            },
+            offset: this.state.offset,
+            limit: this.state.limit,
+            order: "name, id"
+        });
+        this.state.employeesData.records = results.records;
+        this.state.employeesData.count = results.length;
+    }
+
+    async onDepartmentClick(departmentId){
```
Comment (xavierbol) [inline addons/hr_attendance/static/src/components/manual_selection/manual_selection.js:103]:
```suggestion
    async onDepartmentClick(departmentId = false) {
```

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.js [hunk 2]
```diff
@@ -1,59 +1,131 @@
...
+        if (this.env.isSmall) {
+            if (departmentId){
+                const selectedDepartment = this.props.departments.find((department) => department.id === departmentId);
                 this.departmentName = selectedDepartment.name;
+            } else {
+                this.departmentName = _t("All departments");
             }
+        }
+        if (departmentId){
+            this.state.departmentDomain = [['department_id.id', '=', departmentId]];
```
Comment (xavierbol) [inline addons/hr_attendance/static/src/components/manual_selection/manual_selection.js:113]: Equivalent to:

```suggestion
            this.state.departmentDomain = [['department_id', '=', departmentId]];
```

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml [hunk 3]
```diff
@@ -20,31 +20,39 @@
...
+                            onUpdate.bind="_onPagerChanged"/>
+                    </div>
                 </div>
             </div>
             <div class="d-flex h-100 flex-column flex-md-row bg-300">
                 <t t-if="!env.isSmall">
                     <div class="o_hr_kiosk_sidebar ps-2 py-2 overflow-auto">
                         <div class="list-group">
-                            <a t-on-click="() => this.onDepartementClick(false)" class="list-group-item py-3 list-group-item-action text-start">
+                            <a t-on-click="() => this.onDepartmentClick(false)" class="list-group-item py-3 list-group-item-action text-start">
```
Comment (xavierbol) [inline addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml:51]:
````suggestion
                            <a t-on-click="() => this.onDepartmentClick()" class="list-group-item py-3 list-group-item-action text-start">
```

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml [hunk 4]
```diff
@@ -59,15 +67,15 @@
                             <i class="fa fa-caret-down ms-2"/>
                         </button>
                         <t t-set-slot="content">
-                            <a t-on-click="() => this.onDepartementClickMobile(false)" class="d-flex text-start text-reset p-3">
+                            <a t-on-click="() => this.onDepartmentClick(false)" class="d-flex text-start text-reset p-3">
```
Comment (xavierbol) [inline addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml:70]:
```suggestion
                            <a t-on-click="() => this.onDepartmentClick()" class="d-flex text-start text-reset p-3">
```

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.js [hunk 5]
```diff
@@ -1,59 +1,131 @@
...
-            if (selectedDepartment) {
+    async _onPagerChanged({ offset, limit }) {
+        this.state.offset = offset;
+        this.state.limit = limit;
+        await this._fetchEmployeeData();
+    }
+
+    async _fetchEmployeeData() {
+        let domain = this.state.departmentDomain.concat(this.state.searchDomain);
+        let results = await this.orm.webSearchRead("hr.employee", domain, {
```
Comment (xavierbol) [inline addons/hr_attendance/static/src/components/manual_selection/manual_selection.js:87]: Also are we sure the current user has access to the `hr.employee` model?

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left some comments. 🙂

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml [hunk 6]
```diff
@@ -20,31 +20,39 @@
...
                     <div class="o_cp_searchview d-flex input-group h-100">
-                        <div class="o_searchview form-control d-print-contents d-flex align-items-center px-3 rounded-pill border">
-                            <i class="o_searchview_icon d-print-none oi oi-search me-2" aria-label="Search..." title="Search..."/>
-                            <div class="o_searchview_input_container d-flex flex-grow-1 flex-wrap gap-1" style="position: relative;">
-                                <input t-on-input="onSearchInput" type="text" class="o_searchview_input d-print-none flex-grow-1 w-auto border-0" placeholder="Search..."/>
+                        <div class="d-flex flex-row align-items-center rounded-pill border flex-grow-1">
+                            <i class="o_searchview_icon d-print-none oi oi-search m-2" aria-label="Search..." title="Search..."/>
+                            <div class="o_searchview_input_container" style="position: relative;">
+                                <input t-on-input="onSearchInput" type="text" class="d-print-none border-0" style="width: 90%;" placeholder="Search..."/>
```
[the remark this answers is not in the document]
Comment (mepe-odoo) [reply addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml:33]: w-90 doesn't existed :)

[the remark this answers is not in the document]
Comment (xavierbol) [reply addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml:33]: 😢

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml [hunk 7]
```diff
@@ -20,31 +20,39 @@
...
                             </div>
                         </div>
                     </div>
+                    <div class="p-2">
+                        <Pager
+                            t-if="state.employeesData.count > state.limit"
```
[the remark this answers is not in the document]
Comment (mepe-odoo) [reply addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml:39]: I think it's not a good idea i'm a bit stuck now
image

File: addons/hr_attendance/static/src/public_kiosk/public_kiosk_app.js [hunk 8]
```diff
@@ -160,6 +181,7 @@ export async function createPublicKioskAttendance(document, kiosk_backend_info)
     await whenReady();
     const env = makeEnv();
     await startServices(env);
+    session.server_version_info = kiosk_backend_info.server_version_info;
```
Comment (mepe-odoo) [inline addons/hr_attendance/static/src/public_kiosk/public_kiosk_app.js:184]: put this in the next commit

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (xavierbol) [reply addons/hr_attendance/static/src/components/manual_selection/manual_selection.xml:39]: :o

**Review by jugj-odoo — COMMENTED** [review]
Huge work, small comment

**Review by Bertrand2 — CHANGES_REQUESTED** [review]
Not that much to say! Good job :muscle:

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.js [hunk 9]
```diff
@@ -1,59 +1,129 @@
...
+
+    async onDepartmentClick(departmentId = false){
+        if (this.env.isSmall) {
+            if (departmentId){
+                const selectedDepartment = this.props.departments.find((department) => department.id === departmentId);
                 this.departmentName = selectedDepartment.name;
+            } else {
+                this.departmentName = _t("All departments");
             }
+        }
```
[the remark this answers is not in the document]
Comment (mepe-odoo) [reply addons/hr_attendance/static/src/components/manual_selection/manual_selection.js:109]: answer in mp discord

File: addons/hr_attendance/static/src/components/manual_selection/manual_selection.js [hunk 10]
```diff
@@ -1,59 +1,129 @@
...
+        return Math.trunc(availableScreen / employeeCardHeight) * employeeCardPerLine;
     }
 
-    // Changes needed. Can probably be combined with above
-    onDepartementClickMobile(departmentId){
-        if (departmentId) {
-            this.state.displayedEmployees = this.props.employees.filter((item) => item.department.id === departmentId);
-            const selectedDepartment = this.props.departments.find((department) => department.id === departmentId);
-            if (selectedDepartment) {
+    async _onPagerChanged({ offset, limit }) {
```
[the remark this answers is not in the document]
Comment (mepe-odoo) [reply addons/hr_attendance/static/src/components/manual_selection/manual_selection.js:83]: his check seems to be just in python (check_method_name)

File: addons/hr_attendance/static/src/public_kiosk/public_kiosk_app.js [hunk 11]
```diff
@@ -32,6 +35,8 @@ class kioskAttendanceApp extends Component{
             company: this.props.companyId,
         });
         this.state = useState({
+            barcode: false,
```
[the remark this answers is not in the document]
Comment (mepe-odoo) [reply addons/hr_attendance/static/src/public_kiosk/public_kiosk_app.js:38]: in public_kiosk_app.xml line 85 :)

File: addons/web/static/src/views/widgets/documentation_link/documentation_link.xml [hunk 12]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates xml:space="preserve">
     <t t-name="web.DocumentationLink">
-        <a t-att-href="url" title="Documentation" class="o_doc_link me-2" target="_blank">
+        <a t-att-href="url" title="Documentation" t-att-class="classes" target="_blank">
```
[the remark this answers is not in the document]
Comment (mepe-odoo) [reply addons/web/static/src/views/widgets/documentation_link/documentation_link.xml:4]: Jorge jpp from js framework validated it so i don't want to change it :/ and I think it's easier if we want to extends this component

**Comment by Bertrand2** [comment]
robodoo r+

**Comment by Bertrand2** [comment]
robodoo rebase-ff

**Comment by jugj-odoo** [comment]
rebased branch

**Pull request closed without merging**